packages/server/src/auth/clientinfo.ts

```diff
@@ -0,0 +1,24 @@
+import { badRequest } from '@medplum/core';
+import { ClientApplication } from '@medplum/fhirtypes';
+import { Request, RequestHandler, Response } from 'express';
+import { asyncWrap } from '../async';
+import { sendOutcome } from '../fhir/outcomes';
+import { getSystemRepo } from '../fhir/repo';
+import { rewriteAttachments, RewriteMode } from '../fhir/rewrite';
+
+export const clientInfoHandler: RequestHandler = asyncWrap(async (req: Request, res: Response) => {
+  const { clientId } = req.params;
+  const systemRepo = getSystemRepo();
+
+  try {
+    const client = await systemRepo.readResource<ClientApplication>('ClientApplication', clientId);
+    res.status(200).json(
+      await rewriteAttachments(RewriteMode.PRESIGNED_URL, systemRepo, {
+        welcomeString: client.signInForm?.welcomeString,
+        logo: client.signInForm?.logo,
+      })
+    );
+  } catch (_err) {
+    sendOutcome(res, badRequest('Client not found'));
```

Comment:
As mentioned above, you can skip this entire `catch` block.

But, for future reference, the `sendOutcome()` method will set the HTTP status code based on the `OperationOutcome` value.  When using `badRequest`, the HTTP status code will be a generic 400.  If you want a 404 status code, then you need to use the special `notFound` `OperationOutcome`, such as:

```ts
import { notFound } from '@medplum/core`;
```

That's just an FYI.  You can safely delete this line and the `try`/`catch`.